eniius/mcstas.py

```diff
@@ -568,6 +568,8 @@ def mcstasscript_parameter_to_nexus(parameter, index=None):
         return parameter, name
 
     attributes = {f: getattr(parameter, f) for f in fields if hasattr(parameter, f)}
+    # filter out empty-string (or anything equivalent to False) valued attributes since they cause problems :/
+    attributes = {k: v for k, v in attributes.items() if v}
```

Comment:
It must be, but I'm not sure if I didn't add another check!

The extraneous semicolon gives a parameter where all of the checked fields evaluate equivalent to `False` in the if statement. So `attributes` becomes and empty `dict` and `out = NXfield()`.

The slightly worrying-that-it-works bit is that the returned name is `parameter.name` which is `''`, and therefore that the line assigning the collection item becomes `nx[''] = NXfield()`. Does `nexusformat.nexus` do some magic to ignore empty names, or does this overwrite the whole `NXcollection` with an empy `NXfield` object?

Reply:
So, `nx[''] = NXfield()` adds an entry into the `NXcollection` with name `''` -- it doesn't look like the entry can be retrieved via the dictionary syntax (`nv['']` returns the `NXcollection`) but `Writer.to_json` transverses the `.items()` property, so it *is* included in output.

`conv_types` converts the empty `NXfield()` to a string-valued `'None'`, which works but doesn't seem right.
I'll add a check to avoid including the empty parameter into the collection entirely.

Reply:
I've added the check, plus a test case that checks JSON output to ensure no parameters named `''` are produced.